producer.go

```diff
@@ -69,13 +69,12 @@ func (p *Producer) Produce(msg *rdkafka.Message) error {
 // outstanding messages.
 func (p *Producer) Flush(timeoutMs int) int {
 	return p.rdProd.Flush(timeoutMs)
-
 }
 
 // Close stops p after flushing any buffered messages. It is a blocking
 // operation.
 func (p *Producer) Close() {
-	unflushed := p.rdProd.Flush(5000)
+	unflushed := p.Flush(5000)
```

Comment:
How is this change related to the actual issue?

Reply:
It's not

Reply:
That's why it is a separate commit, I can just cherry pick to master before merging, for clarity.

Reply:
Yes that is a great idea